### PR TITLE
add authentication in GraphQL as RestAPI

### PR DESCRIPTION
**Pull request by jbpin**
Graphql was missing authentication as it was already implemented in the Rest API source. This PR add the same options as RestAPI.

### CODE REVIEW

File: plugins/packages/graphql/lib/index.ts [hunk 1]
```diff
@@ -1,46 +1,154 @@
...
+          const headerPrefix = sourceOptions['header_prefix'];
+          headers['Authorization'] = `${headerPrefix}${accessToken}`;
+        }
+      }
+    }
 
     const searchParams = Object.fromEntries(sourceOptions['url_params']);
 
     // Remove invalid entries from the headers and searchParams objects
     Object.keys(headers).forEach((key) => (headers[key] === '' ? delete headers[key] : {}));
```
[the remark this answers is not in the document]
Comment (jbpin) [reply plugins/packages/graphql/lib/index.ts:105]: Hi @shah21 I'm not sure what do you mean as it's the exact implementation that as been done in Rest API plugin (https://github.com/ToolJet/ToolJet/blob/56196930b2840577c61aef196caec04e85f87d56/plugins/packages/restapi/lib/index.ts#L61). I can fix both method but I need more context please.

[the remark this answers is not in the document]
Comment (shah21) [reply plugins/packages/graphql/lib/index.ts:105]: https://github.com/ToolJet/ToolJet/pull/6687/files#diff-d52d6b134c863724fa6f201c15491cfbeaa9c79051da111b37e12da6ba20cc8dR75:~:text=const%20headers%20%3D%20this,headers%27%5D)%2C

I mean we are getting and sanitising the headers with this function right?
and the same function and also the above-marked line is doing the same.

**Comment by shah21** [comment]
other things. I can't see the `Authentication type` dropdown on the `graphql` data source modal 
Screenshot 2023-06-22 at 6 24 35 PM
Screenshot 2023-06-22 at 6 24 58 PM

**Comment by shah21** [comment]
also, I couldn't see the code changes for showing dynamic forms for different authentication types

File: plugins/packages/graphql/lib/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbpin) [reply plugins/packages/graphql/lib/index.ts:105]: @shah21 Ok get it. Thanks

**Comment by jbpin** [comment]
> 

image
I'm not sure if you get the right deployment but here is a screenshot from our. I think field or generated from the manifest

**Comment by shah21** [comment]
Yes. changing the manifest will add components to the UI
But. I tried building plugins and restarting the server and I couldn't see the dropdown.

Also, I didn't find the GitHub issue for this PR. We will probably need a GitHub issue from our side to proceed with this much code changes
Anyway, We will let you know.

**Comment by akshaysasidrn** [comment]
Created an issue for this #6851. Will test it out.

**Comment by akshaysasidrn** [comment]
@jbpin Thanks for the contribution.

**Comment by jbpin** [comment]
@akshaysasidrn you are welcome.

**Pull request merged**